### PR TITLE
asserts: introduce new assertion validation-set

### PR DESCRIPTION
**Pull request by pedronis**
See the doc comment for asserts.ValidationSet.

### CODE REVIEW

File: asserts/validation_set.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	var snapRevision int
+	if _, ok := snap["revision"]; ok {
+		var err error
+		snapRevision, err = checkSnapRevisionWhat(snap, "revision", what)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if snapRevision != 0 && presence == PresenceInvalid {
+		return nil, fmt.Errorf(`cannot specify revision %s at the same time as stating its presence is invalid`, what)
```
Comment (stolowski) [inline asserts/validation_set.go:112]: What do you think a more compact: "cannot specify revision %s and state its presence as invalid" ?

File: asserts/validation_set.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+		snaps = append(snaps, valSetSnap)
+	}
+
+	return snaps, nil
+}
+
+// ValidationSet holds a validation-set assertion, which is a
+// statement by an account about a set snaps and possibly revisions
+// for which an extrisinc/implied property is valid (e.g. they work
```
Comment (stolowski) [inline asserts/validation_set.go:166]: s/extrisinc/extrinsic/

File: asserts/validation_set.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+		return -1, err
+	}
+	if seqnum < 1 {
+		return -1, fmt.Errorf("%q must be >=1: %v", name, seqnum)
+	}
+	return seqnum, nil
+}
+
+var (
+	validValidationSetName = regexp.MustCompile("^[a-z0-9](?:-?[a-z0-9])*$")
```
Comment (stolowski) [inline asserts/validation_set.go:222]: If I read this correctly, a single-character name is OK, but I don't think we have a test case for that below, perhaps it's worth adding one.

File: asserts/validation_set.go (hunk 2 above)
Comment (pedronis) [reply asserts/validation_set.go:166]: I wonder why run-checks didn't catch that

File: asserts/validation_set.go (hunk 1 above)
Comment (pedronis) [reply asserts/validation_set.go:112]: I think the longer version is a bit clearer

File: asserts/validation_set.go (hunk 3 above)
Comment (pedronis) [reply asserts/validation_set.go:222]: we don't tend to have such positive tests, maybe we should have some though for model and validation-set names

Comment (stolowski) [reply asserts/validation_set.go:222]: FWTW we have such tests for snap names (that's where I'm coming from after recent bug concerning 1-character names ;)).
Not super ciritical for this PR but nice to have to make sure we don't accidentally replace "*" with "+" in the future.

**Pull request merged**